test/sqllogictest/chbench.slt

```diff
@@ -1081,20 +1091,20 @@ Explained Query:
     With
       cte l0 =
         Reduce group_by=[#1] aggregates=[sum(#0)] // { arity: 2 }
-          Project (#8, #12) // { arity: 2 }
-            Filter (date_to_timestamp(#6) >= 2007-01-02 00:00:00) // { arity: 13 }
-              Join on=(#4 = #10 AND #5 = #11) type=delta // { arity: 13 }
-                implementation
-                  %0:orderline » %1:stock[#0, #1]UKKA
-                  %1:stock » %0:orderline[#5, #4]KKAif
-                ArrangeBy keys=[[#5, #4]] // { arity: 10 }
-                  Get materialize.public.orderline // { arity: 10 }
-                ArrangeBy keys=[[#0, #1]] // { arity: 3 }
-                  Project (#0, #1, #17) // { arity: 3 }
-                    Get materialize.public.stock // { arity: 18 }
+          Project (#2, #5) // { arity: 2 }
+            Join on=(#0 = #3 AND #1 = #4) type=differential // { arity: 6 }
+              implementation
+                %0:orderline » %1:stock[#0, #1]UKKAif
+              ArrangeBy keys=[[#0, #1]] // { arity: 3 }
+                Project (#4, #5, #8) // { arity: 3 }
+                  Filter (#4) IS NOT NULL AND (#5) IS NOT NULL AND (date_to_timestamp(#6) >= 2007-01-02 00:00:00) // { arity: 10 }
```

Comment:
It seems this Filter was previously being lifted away in order to be able to use the index. Now this is not happening. If you think this would still be good to happen, then maybe you could add the starting input's arrangement to `needed` in `implement_arrangements`.

Reply:
Sorry, maybe I'm misunderstanding something here. I'll study the code a bit more.

Reply:
Ok, so I realized that there is the `order[0].1 = found_keys;`, which should make `implement_arrangements` aware of the need to put `ArrangeBy` on the starting input.

And the lifting is not happening now, because it's a different key, so I guess there is no available index.